crates/starknet/src/lib.rs

```diff
@@ -5,17 +5,20 @@ use blocks::{StarknetBlock, StarknetBlocks};
 use constants::{CHAIN_ID, ERC20_CONTRACT_ADDRESS};
 use predeployed_accounts::PredeployedAccounts;
 use starknet_api::block::{BlockNumber, BlockStatus, BlockTimestamp, GasPrice};
+use starknet_in_rust::core::errors::state_errors::StateError;
 use starknet_in_rust::definitions::block_context::{BlockContext, StarknetOsConfig};
 use starknet_in_rust::definitions::constants::{
     DEFAULT_CAIRO_RESOURCE_FEE_WEIGHTS, DEFAULT_CONTRACT_STORAGE_COMMITMENT_TREE_HEIGHT,
     DEFAULT_GLOBAL_STATE_COMMITMENT_TREE_HEIGHT, DEFAULT_INVOKE_TX_MAX_N_STEPS,
     DEFAULT_VALIDATE_MAX_N_STEPS,
 };
 use starknet_in_rust::execution::TransactionExecutionInfo;
+use starknet_in_rust::state::in_memory_state_reader::InMemoryStateReader;
```

Comment:
Let's create `errors.rs` inside `starknet` crate and initialize there errors for the module/crate. Generally error-enum-per-crate/module pattern shall be used. 
If this error will happen one could think that the error is in `starknet_in_rust` repo

Reply:
Not sure I get the point. Here I am relying on `StateError` defined in our `crates/types/src/error.rs` and it wraps the `starknet_in_rust::core::errors::state_errors::StateError`, so won't it be clear that it's a Devnet error?

On another hand, I assume the return type of my function should be `DevnetResult` (as it already is), but that type requires the Error type to be `crate::error::Error` (the `crate` being `types`).

Reply:
This error can appear in any crate that uses that type. That's why its important to declare errors in every crate and large modules. I created a error enum for starknet crate. It automatically implements from functions

```
#[derive(Error, Debug)]
pub enum Error {
    #[error(transparent)]
    TypesError {
        #[from]
        source: starknet_types::error::Error,
        backtrace: Backtrace,
    },
}
```

Reply:
>  I created a error enum for starknet crate.

Where? In your (unmerged) PR?

Reply:
Will do in my pr